### PR TITLE
Article/fed talk s01e04 react components

### PR DESCRIPTION
**Pull request by qt-pixels**
i am Quintin then master of react

### CODE REVIEW

File: _posts/2021-06-29-fed-talk-s01e04-react-components.md [hunk 1]
```diff
@@ -0,0 +1,545 @@
...
+  * [Children Prop](#children-prop)
+- [:pray: Closing](#pray-closing)
+
+---
+
+## Component Fundamentals
+
+> :warning: React Components can be a sensitive topic, so if you get offended jump to the next chapter.
+
+Components allow us to abstract logic and state away from the the view (DOM). Its existence enables us to think generically and build abstractions, thus, build repeatable testable bits of code.
```
Comment (timveletta) [inline _posts/2021-06-29-fed-talk-s01e04-react-components.md:50]: I would change some of the grammar here:

"Their existence enables us to think generically and build abstractions, allowing for easily testable and reusable bits of code."

File: _posts/2021-06-29-fed-talk-s01e04-react-components.md [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+
+**Long lived the days of re-defining a Button!**
```
Comment (timveletta) [inline _posts/2021-06-29-fed-talk-s01e04-react-components.md:52]: Is it "long lived" or "long gone"?

File: _posts/2021-06-29-fed-talk-s01e04-react-components.md [hunk 3]
```diff
@@ -0,0 +1,545 @@
...
+2. `<input />` field would render
+3. `On Mount` will be logged
+4. Component State gets updated
+5. `Before Render: mounted` will be logged
+6. `<input />` field re-renders
+7. ... something un-mounts the component
+8. `Un Mounting` will be logged
+
+**Note** even though we `setCurrentState()` in the un-mount function, it's side affect is disregarded as the next operation is to destroy that function anyway.
+
```
Comment (timveletta) [inline _posts/2021-06-29-fed-talk-s01e04-react-components.md:189]: As far as I know, you can't access `useState` variables in the un-mount hook but if you needed to, I found this useful blog post: https://www.timveletta.com/blog/2020-07-14-accessing-react-state-in-your-component-cleanup-with-hooks/

**Comment by timveletta** [comment]
As always love your work <3 I feel like its just missing something tying it all together. Like I feel like one of the enjoyable things about working in the same space with you is how you structure and break down your components into understandable chunks and I'm kind of missing that from this post.

I feel like you've got something really important in the section around structuring your components. i.e.
```
* Can it broken down any further? **If yes**, answer the below questions. **If no** and if it does not contain other Components then it is likely a UI Component (low level Component), note it as such and move on.
* Can it be reused anywhere else? **If yes**, make sure your Typed Interface is concrete, **if no**, can you normalize it's implementation to make it re-usable? **If not**, I call that a *bespoke Component*.
* Does it refer to another Component in a similar fashion? **If yes**, is there a way you can generalize the two without bloating the Interface? **If no**, it is likely a new Component.
```

Being able to demonstrate each of these concepts and say here is an example of a bloated component and here is how we can restructure it into several lower level components that are reusable would just tie everything together.

File: _posts/2021-06-29-fed-talk-s01e04-react-components.md (hunk 3 above)
Comment (qt-pixels) [reply _posts/2021-06-29-fed-talk-s01e04-react-components.md:189]: heyyyy **Tim Velettas** website!! hahaha _"I found this useful blog post"_. 
My statement was more around the fact that you `can` access state, but if you try to update internal component state in the unmount phase it'll do nothing.

**Review by timveletta — APPROVED** [review]
Love your work!

**Review by Hamo69 — APPROVED** [review]

**Pull request merged**